### PR TITLE
Fixed #25677 -- Prevented decoding errors in/after Popen calls

### PR DESCRIPTION
**Pull request by claudep**
Just to see what this gives on Windows...

### CODE REVIEW

File: django/core/management/utils.py [hunk 1]
```diff
@@ -10,15 +10,16 @@
...
 
-def popen_wrapper(args, os_err_exc_type=CommandError, universal_newlines=True):
+def popen_wrapper(args, os_err_exc_type=CommandError, stdout_encoding='utf-8'):
     """
     Friendly wrapper around Popen.
 
     Returns stdout output, stderr output and OS status code.
     """
     try:
+        # Using universal_newlines may produce unexpected and annoying decoding errors
```
Comment (timgraham) [inline django/core/management/utils.py]: universal_newlines=False seems to be the default, do we need to be explicit about it? If so, please wrap comment at 79 chars and add period.

File: tests/i18n/test_compilation.py [hunk 2]
```diff
@@ -154,17 +155,23 @@ def test_multiple_locales_excluded_with_locale(self):
...
+        mo_file = 'locale/ja/LC_MESSAGES/django.mo'
+        self.addCleanup(self.rmfile, os.path.join(self.test_dir, mo_file))
         with self.assertRaises(CommandError):
-            call_command('compilemessages', locale=[self.LOCALE], stdout=StringIO())
+            call_command('compilemessages', locale=['ja'], stdout=StringIO())
+
+        # po file contains invalid msgstr content
+        mo_file = 'locale/ko/LC_MESSAGES/django.mo'
+        self.addCleanup(self.rmfile, os.path.join(self.test_dir, mo_file))
+        # Various assertRaises on PY2 don't support unicode error messages
```
Comment (timgraham) [inline tests/i18n/test_compilation.py]: Is it worth using `if six.PY2` and adding the Python 3 version with `assertRaisesMessage` so it's easy to cleanup when dropping PY2?

File: tests/i18n/test_compilation.py [hunk 3]
```diff
@@ -154,17 +155,23 @@ def test_multiple_locales_excluded_with_locale(self):
...
+        try:
+            call_command('compilemessages', locale=['ko'], stdout=StringIO())
+        except CommandError as err:
+            self.assertIn("'�' cannot start a field name", six.text_type(err))
+        except Exception as err:
+            self.fail("'compilemessages' didn't raise CommandError, but %s" % err.__class__.__name__)
```
Comment (timgraham) [inline tests/i18n/test_compilation.py]: We're avoiding the `self.fail()` pattern in favor of letting the entire exception bubble up.

File: tests/i18n/commands/locale/ko/LC_MESSAGES/django.po [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# SOME DESCRIPTIVE TITLE.
```
Comment (timgraham) [inline tests/i18n/commands/locale/ko/LC_MESSAGES/django.po]: Add a descriptive title?

Comment (claudep) [inline tests/i18n/commands/locale/ko/LC_MESSAGES/django.po]: I think we purposefully left unchanged all the default po header values, as this doesn't make any difference (see `git grep "SOME DESCRIPTIVE TITLE" tests`). If we change that one, we should also change all others, including all uppercase values below, not worth it IMHO.

File: tests/i18n/test_compilation.py [hunk 5]
```diff
@@ -154,17 +155,24 @@ def test_multiple_locales_excluded_with_locale(self):
...
+        mo_file = 'locale/ko/LC_MESSAGES/django.mo'
+        self.addCleanup(self.rmfile, os.path.join(self.test_dir, mo_file))
+        if six.PY2:
+            # Various assertRaises on PY2 don't support unicode error messages.
+            try:
+                call_command('compilemessages', locale=['ko'], stdout=StringIO())
+            except CommandError as err:
+                self.assertIn("'�' cannot start a field name", six.text_type(err))
+        else:
+            self.assertRaisesMessage(CommandError, "'�' cannot start a field name")
```
Comment (claudep) [inline tests/i18n/test_compilation.py]: Pfff, obviously!

File: tests/i18n/commands/locale/ko/LC_MESSAGES/django.po (hunk 4 above)
Comment (claudep) [inline tests/i18n/commands/locale/ko/LC_MESSAGES/django.po]: Makes sense

File: tests/i18n/test_compilation.py [hunk 6]
```diff
@@ -154,17 +155,25 @@ def test_multiple_locales_excluded_with_locale(self):
...
-
     def test_error_reported_by_msgfmt(self):
+        # po file contains wrong po formatting.
+        mo_file = 'locale/ja/LC_MESSAGES/django.mo'
+        self.addCleanup(self.rmfile, os.path.join(self.test_dir, mo_file))
         with self.assertRaises(CommandError):
-            call_command('compilemessages', locale=[self.LOCALE], stdout=StringIO())
+            call_command('compilemessages', locale=['ja'], verbosity=0)
+
+        # po file contains invalid msgstr content.
```
Comment (timgraham) [inline tests/i18n/test_compilation.py]: could this be a separate test method? otherwise LGTM

**Pull request merged**